cmd/agent.go

```diff
@@ -24,64 +24,115 @@ func AgentCmd() *cobra.Command {
 		Short: "long running agent for continuously checking policies against plugin data",
 		Long: `The Continuous Compliance Agent is a long running process that continuously checks policy controls
 with plugins to ensure continuous compliance.`,
-		Run: func(cmd *cobra.Command, args []string) {
-			logger := hclog.New(&hclog.LoggerOptions{
-				Name:   "agent",
-				Output: os.Stdout,
-				Level:  hclog.Trace,
-			})
-			pluginRunner := AgentRunner{
-				logger: logger,
-			}
-			err := pluginRunner.Run(cmd, args)
-			if err != nil {
-				log.Fatal(err)
-			}
-		},
+		Run: agentRunner,
 	}
 
 	agentCmd.Flags().StringArray("policy", []string{}, "Directory or Bundle archive where policies are stored")
-	err := agentCmd.MarkFlagRequired("policy")
+	err := agentCmd.MarkFlagDirname("policy")
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("Error marking policy for dirname completion: %v", err)
 	}
 
 	agentCmd.Flags().StringArray("plugin", []string{}, "Plugin executable or directory")
-	agentCmd.MarkFlagsOneRequired("plugin")
+	err = agentCmd.MarkFlagDirname("plugin")
+	if err != nil {
+		log.Fatalf("Error marking plugin for dirname completion: %v", err)
+	}
 
 	agentCmd.Flags().BoolP("daemon", "d", false, "Specify to run as a long running daemon")
 
+	agentCmd.Flags().CountP("verbose", "v", "Enable verbose output")
+
 	return agentCmd
 }
 
-type AgentRunner struct {
-	logger hclog.Logger
+// Main the entrypoint for the `agent` command
+//
+// It reads the policy and plugin paths from the CLI arguments or the config file and then
+// runs the agent process.
+func agentRunner(cmd *cobra.Command, args []string) {
+	config := internal.ReadAgentConfig()
 
-	queryBundles []*rego.Rego
-}
+	verbose, err := cmd.Flags().GetCount("verbose")
+	if err != nil {
+		// Can't use hcl logger as not setup yet
+		log.Fatalf("Error reading verbose flag: %v", err)
+	}
 
-func (ar AgentRunner) Run(cmd *cobra.Command, args []string) error {
-	//ctx := context.TODO()
+	if config != nil {
+		verbose = max(verbose, config.Verbose)
+	}
 
 	policyBundles, err := cmd.Flags().GetStringArray("policy")
 	if err != nil {
-		return err
+		log.Fatalf("Error reading policy flag: %v", err)
+	}
+
+	if len(policyBundles) == 0 && config != nil {
```

Comment:
Is this correct? Feels like it should be `if len(policyBundles) > 0 && config != nil`

Reply:
This is outdated now thanks to the latest viper work.